File: Temp/leetcode/editor/cn/[1422]分割字符串的最大得分.c
// 1422 分割字符串的最大得分
// https://leetcode.cn/problems/maximum-score-after-splitting-a-string/


//leetcode submit region begin(Prohibit modification and deletion)
#define MAX(a, b) ((a) > (b) ? (a) : (b))
int maxScore(char* s) {
    int n = strlen(s);
    int one = 0, zero = 0;
    for (int i = 0; i < n; i++)
        one += s[i] == '1';
    int ans = 0;
    for (int i = 0; i < n - 1; i++) {
        zero += s[i] == '0';
        one -= s[i] == '1';
        ans = MAX(ans, one + zero);
    }
    return ans;
}
//leetcode submit region end(Prohibit modification and deletion)

File: Temp/leetcode/editor/cn/[1422]分割字符串的最大得分.py
# 1422 分割字符串的最大得分
# https://leetcode.cn/problems/maximum-score-after-splitting-a-string/


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def maxScore(self, s: str) -> int:
        one = s.count('1')
        zero = 0
        ans = 0
        for c in s[:-1]:
            zero += c == '0'
            one -= c == '1'
            ans = max(ans, zero + one)
        return ans
# leetcode submit region end(Prohibit modification and deletion)
